modules/cloudwatch-logs/providers.tf

```diff
@@ -1,22 +1,36 @@
 provider "aws" {
   region = var.region
 
-  # `terraform import` will not use data from a data source, so on import we have to explicitly specify the profile
-  profile = coalesce(var.import_profile_name, module.iam_roles.terraform_profile_name)
+  profile = module.iam_roles.profiles_enabled ? coalesce(var.import_profile_name, module.iam_roles.terraform_profile_name) : null
 
-  # See issue: https://github.com/hashicorp/terraform-provider-aws/issues/18311
-  # default_tags {
-  #   tags = module.this.tags
-  # }
+  dynamic "assume_role" {
+    for_each = module.iam_roles.profiles_enabled ? [] : ["role"]
+    content {
+      role_arn = coalesce(var.import_role_arn, module.iam_roles.terraform_role_arn)
+    }
+  }
 }
 
 module "iam_roles" {
-  source  = "../account-map/modules/iam-roles"
-  context = module.this.context
+  source             = "../account-map/modules/iam-roles"
+  global_tenant_name = var.root_account_tenant_name
```

Comment:
we can remove `global_tenant_name`, as it's moved to `account-map/modules/iam-roles`